pydeseq2/dds.py

```diff
@@ -88,63 +98,49 @@ class DeseqDataSet:
 
     Attributes
     ----------
-    design_matrix : pandas.DataFrame
-        A DataFrame with experiment design information (to split cohorts).
-        Indexed by sample barcodes. Unexpanded, *with* intercept.
-
-    n_processes : int
-        Number of cpus to use for multiprocessing.
-
-    size_factors : pandas.Series
-        DESeq normalization factors.
-
-    non_zero_genes : pandas.Index
-        Index of genes that have non-uniformly zero counts.
-
-    genewise_dispersions : pandas.Series
-        Initial estimates of gene counts dispersions.
+    X
+        A ‘number of samples’ x ‘number of genes’ count data matrix.
 
-    trend_coeffs : pandas.Series
-        Coefficients of the trend curve: :math:`f(\mu) = \alpha_1/ \mu + a_0`.
+    obs
+        Key-indexed one-dimensional observations annotation of length 'number of
+        samples". Used to store design factors.
 
-    fitted_dispersions : pandas.Series
-        Genewise dispersions regressed on the trend curve.
+    var
+        Key-indexed one-dimensional gene-level annotation of length ‘number of genes’.
 
-    prior_disp_var : float
-        Dispersion prior of genewise dispersions,
-        used for dispersion shrinkage towards the trend curve.
+    uns
+        Key-indexed unstructured annotation.
 
-    MAP_dispersions : pandas.Series
-        MAP dispersions, after shrinkage towards the trend curve and before filtering.
+    obsm
+        Key-indexed multi-dimensional observations annotation of length
+        ‘number of samples’. Stores "design_matrix" and "size_factors", among others.
 
-    dispersions : pandas.Series
-        Final dispersion estimates, after filtering MAP outliers.
+    varm
+        Key-indexed multi-dimensional gene annotation of length ‘number of genes’.
+        Stores "dispersions" and "LFC", among others.
 
-    LFCs : pandas.DataFrame
-        Log-fold change and intercept parameters, in natural log scale.
+    layers
+        Key-indexed multi-dimensional arrays aligned to dimensions of `X`, e.g. "cooks".
 
-    cooks : pandas.DataFrame
-        Cooks distances, used for outlier detection.
+    n_processes : int
+        Number of cpus to use for multiprocessing.
 
-    replaceable : pandas.Series
-        Whether counts are replaceable, i.e. if a given condition has enough samples.
+    non_zero : ndarray
+        Boolean mask of genes that have non-uniformly zero counts.
```

Comment:
Why not store it under `varm`? If I understand correctly, this variable would have the correct size and it is its purpose somehow?

Reply:
Done